src/Directory.Build.targets

```diff
@@ -2,16 +2,63 @@
 
   <Import Project="Sdk.targets" Sdk="Microsoft.DotNet.Arcade.Sdk" />
 
+  <PropertyGroup>
+    <TestRoot Condition="'$(TestRoot)' == ''">$(ArtifactsObjDir)generatedtests/</TestRoot>
+  </PropertyGroup>
+
+  <Target Name="CleanTestRoot">
+    <!-- Clean the test root to avoid dirty content from a previous run -->
+    <RemoveDir Directories="$(TestRoot)" />
+  </Target>
+
+  <Target Name="CreateDirectoryBuildFiles">
+    <!-- Define Directory.Build.* files in the scenario tests artifacts directory to prevent the test run from
+          picking up the configuration from the VMR's Directory.Build.* files. We need an isolated configuration
+          from which to test. -->
+    <WriteLinesToFile File="$(TestRoot)Directory.Build.targets"
+                      Lines="&lt;Project /&gt;"
+                      Overwrite="true" />
+    <!-- Because the VMR (https://github.com/dotnet/dotnet) is configured to use CPM (https://github.com/dotnet/installer/pull/19286), it will
+          cause the scenario tests to fail with the following error:
+            Projects that use central package version management should not define the version on the PackageReference items but on the PackageVersion items: ...
+          To work around this, we explicitly disable CPM. Note that this can't be solved by including a stub Directory.Packages.props file because the default
+          behavior is to automatically default ManagePackageVersionsCentrally to true when it finds such a file.
+          https://github.com/NuGet/NuGet.Client/blob/ca13cf0a281b9774dd0238a43ab98c1927056cc2/src/NuGet.Core/NuGet.Build.Tasks/NuGet.props#L25-L33 -->
+    <PropertyGroup>
+      <_DirectoryBuildPropsContent>
+&lt;Project&gt;
+  &lt;PropertyGroup&gt;
+    &lt;ManagePackageVersionsCentrally&gt;
+      false
+    &lt;/ManagePackageVersionsCentrally&gt;
+  &lt;/PropertyGroup&gt;
+&lt;/Project&gt;
+      </_DirectoryBuildPropsContent>
+    </PropertyGroup>
+    <WriteLinesToFile File="$(TestRoot)Directory.Build.props"
+                      Lines="$(_DirectoryBuildPropsContent)"
+                      Overwrite="true" />
+  </Target>
```

Comment:
Just saw this PR. I think it might be cleaner to just set the following env vars or properties in the generated projects: `ImportDirectoryBuildProps=false` `ImportDirectoryBuildTargets=false` and `ImportDirectoryPackagesProps=false`. That avoids creating these intermediate files and removes the need for this target.

We do that for the smoke tests that also create intermediate projects: https://github.com/dotnet/dotnet/blob/212c615a6283f3904a2e78b1a56d30363bc11634/test/Microsoft.DotNet.SourceBuild.SmokeTests/DotNetHelper.cs#L133-L136

Reply:
@mthalman just double checking, did you follow-up on this feedback or do we have a tracking issue for it?

Reply:
Sorry, no, I missed that. I've logged https://github.com/dotnet/source-build/issues/4489.